File: tests/fixtures/vulnerable.js
// Deliberately vulnerable JavaScript file for testing Anty
// DO NOT use any of this code in production!

const AWS_KEY = "AKIAIOSFODNN7EXAMPLE";
const AWS_SECRET = process.env.AWS_SECRET; // This one is fine

// Bad: hardcoded database password
const DB_PASSWORD = "super_secret_password_123!";

// Bad: database connection string with credentials
const mongoUrl = "mongodb://admin:p4ssw0rd@db.example.com:27017/myapp";

// Bad: GitHub token
const token = "ghp_ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghij";

// Bad: Stripe key
const stripeKey = "sk_live_abc123def456ghi789jkl012mno";

// Bad: eval usage
function handleUserInput(input) {
    return eval(input);
}

// Bad: SQL injection via string concatenation
function getUser(userId) {
    const query = "SELECT * FROM users WHERE id = " + userId;
    return db.execute(query);
}

// Bad: SQL injection via template literal
function findByEmail(email) {
    return db.query(`SELECT * FROM accounts WHERE email = '${email}'`);
}

// Bad: innerHTML
function renderContent(html) {
    document.getElementById("output").innerHTML = html;
}

// Bad: dangerouslySetInnerHTML
function RawHtml({ content }) {
    return <div dangerouslySetInnerHTML={{ __html: content }} />;
}

// Bad: MD5 usage
const crypto = require('crypto');
const hash = crypto.createHash('md5').update(data).digest('hex');

// Bad: CORS wildcard
app.use(cors({ origin: '*' }));

// Bad: Debug mode
const config = { DEBUG: true, port: 3000 };

// Bad: SSL verification disabled
const agent = new https.Agent({ rejectUnauthorized: false });

// Bad: hardcoded JWT secret
const JWT_SECRET = "my_super_secret_jwt_key_do_not_share";

// Good: using environment variables (should NOT trigger)
const apiKey = process.env.API_KEY;
const dbUrl = process.env.DATABASE_URL;

File: tests/fixtures/vulnerable.py
# Deliberately vulnerable Python file for testing Anty
# DO NOT use any of this code in production!

import pickle
import yaml
import subprocess
import hashlib
import os

# Bad: hardcoded AWS key
AWS_ACCESS_KEY = "AKIAIOSFODNN7EXAMPLE"

# Bad: hardcoded password
db_password = "admin123!"

# Bad: database URL with credentials
DATABASE_URL = "postgres://dbuser:s3cur3p4ss@db.prod.example.com:5432/production"

# Bad: private key in source
PRIVATE_KEY = """-----BEGIN RSA PRIVATE KEY-----
MIIEpAIBAAKCAQEA0Z3VS5JJcds3xfn/ygWyF8PbnGcY5unA67hbYMC+...
-----END RSA PRIVATE KEY-----"""

# Bad: pickle deserialization
def load_data(data):
    return pickle.loads(data)

# Bad: unsafe yaml
def load_config(path):
    with open(path) as f:
        return yaml.load(f.read())

# Bad: shell=True
def run_command(user_input):
    subprocess.call(f"echo {user_input}", shell=True)

# Bad: SQL injection via f-string
def get_user(cursor, user_id):
    cursor.execute(f"SELECT * FROM users WHERE id = {user_id}")

# Bad: MD5 hashing
def hash_password(password):
    return hashlib.md5(password.encode()).hexdigest()

# Bad: eval
def calculate(expression):
    return eval(expression)

# Good: proper environment variable usage (should NOT trigger)
API_KEY = os.environ.get("API_KEY")
DB_HOST = os.getenv("DB_HOST", "localhost")
